### PR TITLE
Add ipl3 astra paths and new IPL-3 config

### PR DESCRIPTION
**Pull request by andycasey**

### CODE REVIEW

**Comment by havok2063** [comment]
In addition to updated paths in `sdsswork.cfg`, there should be a new `ipl3.cfg` with all the paths and envvars defined there.  If using IPL-2 as a base config, then only new or updated paths and envars needs to be included.

**Comment by havok2063** [comment]
See https://github.com/sdss/tree/issues/65

**Comment by joelbrownstein** [comment]
We would need the ipl3.cfg after that top level folder has been created on the SAS.  We are currently staging ipl3 files in sdsswork, and since there are therefore no current paths starting with ipl3, then this action should be deferred until the sdsswork -> ipl3 copy is scheduled and completed.

**Comment by andycasey** [comment]
OK. When the ipl3.cfg file is ready just @/ping me through some electronic medium and I will add these paths to the ipl3.cfg file. After discussion with @ajwheeler I have corrected some of these paths (I am an idiot who often makes mistakes). I think the paths are all correct now.

**Review by havok2063 — DISMISSED** [review]

File: data/ipl3.cfg [hunk 1]
```diff
@@ -0,0 +1,296 @@
...
+astraVisitSLAM = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/results/visit/@cat_id_groups|/astraVisit-SLAM-{v_astra}-{cat_id}@component_default|-{task_id}.fits
+astraVisitWD = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/results/visit/@cat_id_groups|/astraVisit-WD-{v_astra}-{cat_id}@component_default|-{task_id}.fits
+
+astraAllStarAPOGEENet  = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-ASPOGEENet-{v_astra}-{run2d}-{apred}.fits
+astraAllStarASPCAP     = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-ASPCAP-{v_astra}-{run2d}-{apred}.fits
+astraAllStarCannon     = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-Cannon-{v_astra}-{run2d}-{apred}.fits
+astraAllStarClassifier = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-Classifier-{v_astra}-{run2d}-{apred}.fits
+astraAllStarThePayne   = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-Payne-{v_astra}-{run2d}-{apred}.fits
+astraAllStarSLAM       = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-SLAM-{v_astra}-{run2d}-{apred}.fits
+astraAllStarSnowWhite  = $MWM_ASTRA/{v_astra}/{run2d}-{apred}/summary/astraAllStar-SnowWhite-{v_astra}-{run2d}-{apred}.fits
```
Comment (ajwheeler) [inline data/ipl3.cfg:156]: I think these are redundant?

**Comment by ajwheeler** [comment]
Otherwise lgtm

**Comment by joelbrownstein** [comment]
Hi @andycasey @ajwheeler @havok2063, I added a top-level ipl3 folder at /uufs/chpc.utah.edu/common/home/sdss/ipl-3, or on the SAS athttps://data.sdss5.org/sas/ipl-3; and I added the ipl3.cfg file and copied over boss_drp, apogee_drp, and astra paths from sdsswork.  Would you please check that the paths are OK in this branch so we can merge to main and install at Utah?

**Review by havok2063 — CHANGES_REQUESTED** [review]
The build/test action is failing.  As @ajwheeler noticed, there are redundant, duplicate, paths in the `ipl3.cfg`.  I think these are the cause of the failing action.   

Because the scope of this PR has now expanded, additional updates need to be added to the documentation and tests before we can merge this PR and tag a new release of tree.  Can you update the relevant rst files: `docs/sphinx/config.rst` and `tree_envs.rst`?    

Can you also add a small test in `test_tree.py` that loads the IPL-3 config and checks that a correct IPL-3 specific environment and path template are there?

**Comment by joelbrownstein** [comment]
I added the test @havok2063 requested, but @andycasey or @ajwheeler would you please remove the duplicates from sdssword and ipl3 -- I don't know which to keep since the duplicates don't match!

**Comment by havok2063** [comment]
Currently I cannot read the ipl3 config in tree with the following error:  This is a straight error from `ConfigParser` reading in an ini file.  The ini file can't have duplicate named keys in any section. 
```
DuplicateOptionError: While reading from '/Users/Brian/Work/github_projects/sdss/tree/data/ipl3.cfg' [line 159]: option 'allCal' in section 'PATHS' already exists
```
I think large chunks of paths were just copied over for boss, apogee, and astra.

**Comment by joelbrownstein** [comment]
I made an attempt to remove the duplicates.

**Comment by havok2063** [comment]
@ajwheeler Can you review the updated ipl3 config and just double check that everything looks ok after the cleanup of the duplicates?

**Review by havok2063 — APPROVED** [review]

**Comment by ajwheeler** [comment]
Looks like the deduplication is correct.

**Pull request merged**